util/patch/patch_test.go

```diff
@@ -918,3 +919,75 @@ func TestNewHelperNil(t *testing.T) {
 	_, err = NewHelper(nil, nil)
 	g.Expect(err).NotTo(BeNil())
 }
+
+func Test_sanitizePatch(t *testing.T) {
+	tests := []struct {
+		name    string
+		diffMap map[string]interface{}
+		paths   []string
+		want    map[string]interface{}
+	}{
+		{
+			name: "Simple patch",
+			diffMap: map[string]interface{}{
+				"spec": map[string]interface{}{
+					"joinConfiguration": "config",
+				},
+			},
+			paths: []string{"spec.joinConfiguration"},
+			want: map[string]interface{}{
+				"spec": map[string]interface{}{
+					"joinConfiguration": "redacted",
+				},
+			},
+		},
+		{
+			name: "Realistic object patch",
```

Comment:
Let's assume I want to redact a whole sub-structure. Then I would pass in `spec.joinConfiguration` and would expect that `spec.joinConfiguration` is set to redacted. Can we add a test for this?

Another thing we should test. Let's assume we have a redacted path which is not a part of the patch. I would expect that the patch is not modified. Can we add a test case for that as well?

Reply:
I'll add a test, but that's the intended (and if I'm right current :) ) behaviour. It replaces the entire substructure with "redacted"

Reply:
Yup. I didn't really think through the existing code :). Just looked at it from the test perspective and wanted to have both cases I mentioned covered.